src/Configuration/AutoMapperConfig.php

```diff
@@ -202,11 +227,64 @@ public function registerMapping(
         return $mapping;
     }
 
+
+    public function registerMiddlewares(Middleware ...$middlewares): AutoMapperConfigInterface
```

Comment:
This way you can only register all middlewares at once. It might be interesting to add an `addPropertyMiddleware` and `addMapperMiddleware` method. Or maybe we can take a look at [Guzzle's way of doing things](https://github.com/guzzle/guzzle/blob/master/src/HandlerStack.php) and make use of a separate stack class.

Reply:
Ok, i'll change this ! Sorry for the delay, i'm busy on one project right now, but I use this current fork for it and it really solves many use cases :).

I'll try to update the pull request tomorrow.

Reply:
No problem at all; I took a long time to reply myself :)

Reply:
@mark-gerarts I'm reading this pull request again, and it's not required to register all middlewares at once, because registerMiddlewares is chainable, and invoking registerMiddleswares won't remove already registered middlewares.

So you can add middlewares one by one, by calling registerMiddlewares as many times as you want.

Do you plan to merge this PR and release 2.0 ?

Reply:
Yes, I want to include this in release 2.0, which I aim to release this summer. As for the adding of middlewares: registerMiddlewares does indeed allow for chaining, but maybe It's useful to add some utility methods such as add. I'll look into this into detail once I start merging the PR.

Reply:
Hi @mark-gerarts, any ETA for this release ?

Reply:
I've set up [a board](https://github.com/mark-gerarts/automapper-plus/projects/1) for the release. The scope is rather limited, but my time is as well at the moment. To be honest, I have not really been able to work on the release the past months.

I *should* be able to finally work a bit on the release again the following weeks (:crossed_fingers:). The question is if it will be enough to finish all open tasks. I'll keep you posted once I've worked myself in again and have a better overview!